From d96e9a40e9da9163e0a4b3475178fe30c26deb19 Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Fri, 22 Sep 2023 22:59:56 +0000
Subject: [PATCH] Musl does not have stack unwinder like glibc therefore we can

 not assume that its always available on musl, we do need to check for target
 environment as well which could be musl or glibc.

Upstream-Status: Pending
Signed-off-by: Khem Raj <raj.khem@gmail.com>
---
 mozglue/misc/StackWalk.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/mozglue/misc/StackWalk.cpp b/mozglue/misc/StackWalk.cpp
index 18fd3464b0..d3d9208fdb 100644
--- a/mozglue/misc/StackWalk.cpp
+++ b/mozglue/misc/StackWalk.cpp
@@ -50,7 +50,7 @@ using namespace mozilla;
 #  define HAVE___LIBC_STACK_END 0
 #endif
 
-#if (defined(linux) &&                                            \
+#if (defined(linux) && defined(__GLIBC__) &&                      \
      ((defined(__GNUC__) && (defined(__i386) || defined(PPC))) || \
       defined(HAVE__UNWIND_BACKTRACE)) &&                         \
      (HAVE___LIBC_STACK_END || ANDROID))
